src/buildercore/bootstrap.py

```diff
@@ -96,6 +97,8 @@ def _create_generic_stack(stackname, parameters=None, on_start=_noop, on_error=_
         context = context_handler.load_context(stackname)
         # setup various resources after creation, where necessary
         setup_ec2(stackname, context['ec2'])
+        # since setup_ec2 doesn't actually set up, this may have to be moved
```

Comment:
there is no reason to wait until EC2 is up, so can be done as soon as `_wait_until_in_progress` has confirmed CloudFormation has completed the creation of the stack. I wouldn't run Terraform and CloudFormation in parallel for simplicity.